authority/voting/client/client.go

```diff
@@ -162,17 +163,22 @@ func (p *connector) initSession(ctx context.Context, doneCh <-chan interface{},
 
 	// Initialize the wire protocol session.
 	var ad []byte
+	var pkiSignatureScheme sign.Scheme
 	if signingKey != nil {
 		keyHash := hash.Sum256From(signingKey)
 		ad = keyHash[:]
+		pkiSignatureScheme = signSchemes.ByName(signingKey.Scheme().Name())
+	} else {
+		pkiSignatureScheme = signSchemes.ByName("Ed25519 Sphincs+") // todo fix me
```

Comment:
it should obtain the signature scheme from the config file

Reply:
if signingKey is allowed to be nil then you should not be deriving the scheme from it. instead get the signing scheme from the config.